javav2/example_code/transcribe-streaming/src/main/java/com/amazonaws/transcribestreaming/StreamingRetryApp.java

```diff
@@ -1,95 +1,95 @@
-                        new FileInputStream(new File("FileName"))),
-                /**
-                 * Object that defines the behavior on how to handle the stream
-                 */
-                new StreamTranscriptionBehaviorImpl());
-
-        /**
-         * Synchronous wait for stream to close, and close client connection
-         */
-        result.get();
-        client.close();
-    }
-
-    private static class AudioStreamPublisher implements Publisher<AudioStream> {
-        private final InputStream inputStream;
-        private static Subscription currentSubscription;
-
-        private AudioStreamPublisher(InputStream inputStream) {
-            this.inputStream = inputStream;
-        }
-
-        @Override
-        public void subscribe(Subscriber<? super AudioStream> s) {
-            if (this.currentSubscription == null) {
-                this.currentSubscription = new TranscribeStreamingDemoApp.SubscriptionImpl(s, inputStream);
-            } else {
-                this.currentSubscription.cancel();
-                this.currentSubscription = new TranscribeStreamingDemoApp.SubscriptionImpl(s, inputStream);
-            }
-            s.onSubscribe(currentSubscription);
-        }
-    }
-}
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazonaws.transcribestreaming;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.transcribestreaming.model.AudioStream;
+import software.amazon.awssdk.services.transcribestreaming.model.LanguageCode;
+import software.amazon.awssdk.services.transcribestreaming.model.StartStreamTranscriptionRequest;
+import javax.sound.sampled.LineUnavailableException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+// snippet-start:[transcribe.java-streaming-retry-app]
+public class StreamingRetryApp {
+    private static final String endpoint = "endpoint";
+    private static final Region region = Region.US_EAST_1;
+    private static final int sample_rate = 28800;
+    private static final String encoding = " ";
+    private static final String language = LanguageCode.EN_US.toString();
+
+    public static void main(String args[]) throws URISyntaxException, ExecutionException, InterruptedException,
+            LineUnavailableException, FileNotFoundException {
+        /**
+         * Create Amazon Transcribe streaming retry client.
+         */
+
+        TranscribeStreamingRetryClient client = new TranscribeStreamingRetryClient(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *client*. The resource is referenced at line 51. The resource is closed at line 72. There are other execution paths that do not contain closure statements, for example, when FileInputStream constructor throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).